src/methods.jl

```diff
@@ -179,3 +179,71 @@ function trimview(frame::AbstractArray, idxs)
 
     return selectdim(frame, d, complement_idxs)
 end
+
+
+"""
+    crop(frame::AbstractArray, shape; force_equal = true)
+
+Crops the image to the shape specified asumming center as the anchor.
+
+Shape speficies the size of the cropped array, force_equal when set to true forces the cropped rows and columns to be equal on both sides
+otherwise the cropped frame contains one extra column on left and one extra row on top if necessary.
+
+# Examples
+```jldoctest
+julia> frame = reshape(1:25, (5, 5));
+
+julia> crop(frame, (3, 3))
+3×3 Array{Int64,2}:
+ 7  12  17
+ 8  13  18
+ 9  14  19
+
+```
+
+# See Also
+* [`cropview`](@ref)
+"""
+crop(frame::AbstractArray, shape; kwargs...) = copy(cropview(frame, shape; kwargs...))
+
+
+"""
+    cropview(frame::AbstractArray, shape; force_equal = true)
+
+Crops the image to the shape specified asumming center as the anchor.
+
+This function is same as the [`crop`](@ref) function but returns a view of the frame.
+
+!!! note
+    This function returns a view of the frame, so any modification to output
+    array will result in modification of frame.
+
+# See Also
+* [`crop`](@ref)
+"""
+function cropview(frame::AbstractArray, shape; force_equal = true)
+    shape_diff = size(frame) .- shape
+
+    # testing error
+    any(s -> s < 0, shape_diff) && error("crop size ($shape) can't be larger than frame size ($(size(frame)))")
+    any(s -> s < 0, shape) && error("crop size ($shape) cant't be negative")
+
+    # calculating range of cropped frame
+    range = []
```

Comment:
I'd initialise with
```suggestion
    range = Vector{Int}(undef, 2)
```
and use assignment below, rather than `push!`ing

Reply:
This should probably be: `range = Vector{Vector{Int}}(undef, 2)` because we are storing ranges in this.

Reply:
Then
```suggestion
    range = Vector{UnitRange{Int}}(undef, 2)
```

Reply:
Switched to `maps` in place of `for` loop!